File: admin/get cli/e_bandobas/lib/utils/text_utils.dart
import 'package:e_bandobas/app/Exceptions/custom_exception.dart';

class TextUtils {

    /**
     * Returns true if the parameter is null or of zero length
     */
    static bool isEmpty(final String? s) {
        if (s == null) {
            return true;
        }
        return s.isEmpty;
    }

    static bool notBlankNotEmpty(String? s) {
        if (s == null) {
            return false;
        }
        return s.trim().isNotEmpty;
    }
    
    static String notBlankNotEmptyWithDefault(String? s, String defaultValue){
        if (s == null){
            return defaultValue;
        }
        return s.trim().isEmpty ? defaultValue : s;
    }
    /**
     * @since 4.4
     */
    static bool containsBlanks(final String? s) {
        if (s == null) {
            return false;
        }
        for (int i = 0; i < s.length; i++) {
            if (s[i] == " ") {
                return true;
            }
        }
        return false;
    }

    static String notBlankNotEmptyOrError(String s, String errorMsg) {
        if (notBlankNotEmpty(s)){
            return s;
        }
        throw CustomException(errorMsg);
    }

}

File: gujarat_police_backend/src/main/java/com/shyam/gujarat_police/util/TextUtils.java
package com.shyam.gujarat_police.util;


import com.shyam.gujarat_police.exceptions.CustomException;

/**
 * @since 4.3
 */
public final class TextUtils {

    /**
     * Returns true if the parameter is null or of zero length
     */
    public static boolean isEmpty(final CharSequence s) {
        if (s == null) {
            return true;
        }
        return s.length() == 0;
    }

    /**
     * Returns true if the parameter is null or contains only whitespace
     */
    public static boolean isBlank(final CharSequence s) {
        if (s == null) {
            return true;
        }
        for (int i = 0; i < s.length(); i++) {
            if (!Character.isWhitespace(s.charAt(i))) {
                return false;
            }
        }
        return true;
    }

    public static boolean notBlankNotEmpty(String s) {
        if (s == null) {
            return false;
        }
        return s.trim().length() != 0;
    }
    public static String notBlankNotEmptyWithDefault(String s, String defaultValue){
        if (s == null){
            return defaultValue;
        }
        return s.trim().length() == 0 ? defaultValue : s;
    }
    /**
     * @since 4.4
     */
    public static boolean containsBlanks(final CharSequence s) {
        if (s == null) {
            return false;
        }
        for (int i = 0; i < s.length(); i++) {
            if (Character.isWhitespace(s.charAt(i))) {
                return true;
            }
        }
        return false;
    }

    public static String notBlankNotEmptyOrError(String s, String errorMsg) {
        if (notBlankNotEmpty(s)){
            return s;
        }
        throw new CustomException(errorMsg);
    }

}

